### PR TITLE
Add k8s integration

### PR DESCRIPTION
**Pull request by jswoods**
* Adds k8s integration
* Adds test options
* Uses godep restore and removes vendor deps

### CODE REVIEW

File: roller.go [hunk 1]
```diff
@@ -576,6 +658,10 @@ func main() {
 
 	wg.Wait()
 
+	if state.clusterAutoscaler.enabled == true {
+		enableClusterAutoscaler(state)
```
Comment (someword) [inline roller.go]: the comparison to true can be left off with the same affect

if state.clusterAutoscaler.enabled {

File: bin/run-test-suite.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -ex
+
+godep go test -v \
```
Comment (jswoods) [inline bin/run-test-suite.sh]: No, it captures the roller.go code and then breaks the test.

File: client_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+func NewFakeClient() KubernetesClient {
+	return &FakeKubernetesClientConfig{}
+}
+
+func (c FakeKubernetesClientConfig) getDeployment(service string, namespace string) (*v1beta1.Deployment, error) {
+	if service == fakeDeployment.Spec.Template.ObjectMeta.Name && namespace == fakeDeployment.Spec.Template.ObjectMeta.Namespace {
+		return fakeDeployment, nil
+	} else {
+		error := errors.New(fmt.Sprintf("deployments.extensions \"%s\" not found", service))
```
Comment (jswoods) [inline client_test.go]: Ah, didnt know fmt had this option. Will change

File: deployment_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+func TestGetReplicaCount(t *testing.T) {
+	client := NewFakeClient()
+	deploymentController := KubernetesDeployment{service: "fake-service",
+		namespace: "fake-namespace"}
+	deploymentObject, _ := deploymentController.GetDeployment(client)
+	replicas := *deploymentObject.Spec.Replicas
+	if replicas != int32(1) {
+		t.Error(fmt.Sprintf("expected 1, got %i", replicas))
```
Comment (jswoods) [inline deployment_test.go]: Nothing. Typo

File: roller.go (hunk 1 above)
Comment (jswoods) [inline roller.go]: Good call. Will fix

**Pull request merged**